lib/portage/dbapi/bintree.py

```diff
@@ -1234,11 +1234,32 @@ def _populate_local(self, reindex=True):
 
         return pkgindex if update_pkgindex else None
 
+    def _run_trust_helper(self):
+        portage_trust_helper = self.settings.get("PORTAGE_TRUST_HELPER", "true")
+        # getuto is a shell script...
+        ret = os.waitstatus_to_exitcode(os.system(portage_trust_helper))
+        if ret == 127:
+            raise OSError(
```

Comment:
Application code should almost never raise `OSError`. This is only meant to be raised from system calls.

Reply:
So what hell should I raise instead?

Reply:
I see OSError being raised elsewhere in this file, so I guess it's ok.

A more specific option would be [FileNotFoundError](https://docs.python.org/3/library/exceptions.html#FileNotFoundError), which would normally be raised if `subprocess.run` cannot find the program to be executed. Using `shell=True` changes the behavior because we get bash's return code instead.

Reply:
If we wanted a more application-specific exception, it looks like common exception classes are defined in lib/portage/exception.py. Several of them are imported at the top of this file.